app/src/main/java/io/appium/settings/receivers/UnpairBluetoothDevicesReceiver.java

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings.receivers;
+
+import android.app.Activity;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import java.util.Set;
+
+public class UnpairBluetoothDevicesReceiver extends BroadcastReceiver implements HasAction {
+    private static final String TAG = UnpairBluetoothDevicesReceiver.class.getSimpleName();
+
+    private static final String ACTION = "io.appium.settings.unpair_bluetooth";
+    private boolean isFailed = false;
```

Comment:
what is the meaning of this property? Why it should be an instance-level variable?

Reply:
After extracting unpairBluetoothDevices(Set<BluetoothDevice> pairedDevices) I can just setResultCode() in there, so it will be no longer needed.

```
private void unpairBluetoothDevices(Set<BluetoothDevice> pairedDevices) {
        for (BluetoothDevice device : pairedDevices) {
            try {
                device.getClass().getMethod("removeBond", null).invoke(device, null);
            } catch (Exception e) {
                setResultCode(Activity.RESULT_CANCELED);
                Log.e("UnpairBluetooth", String.format("Unpairing bluetooth device %s failed. %s", device, e.getMessage()));
                return;
            }
        }
        setResultCode(Activity.RESULT_OK);
    }
```

Reply:
^ this solution would make more sense.

Reply:
done